File: hang.liu/746.js
var minCostClimbingStairs = function(cost) {
  var result = cost.slice();
  result.unshift(0);

  var length = result.length;

  for (var i = 2; i < length; i++) {
    result[i] = result[i] + Math.min(result[i - 1], result[i - 2]);
  }

  return Math.min(result[length - 1], result[length - 2]);
};

File: xiang.li/746.kt
import kotlin.math.min
class Solution {
    fun minCostClimbingStairs(cost: IntArray): Int {
        for(i in 2 until cost.size){
            cost[i] += min(cost[i - 1], cost[i - 2])
        }
        return min(cost[cost.size - 1], cost[cost.size - 2])
    }
}
